platform/include/platform/CircularBuffer.h

```diff
@@ -335,9 +337,23 @@ class CircularBuffer {
         return elements;
     }
 
-    CounterType incrementCounter(CounterType val)
+    /** Used to increment _tail or _head by a given value.
+     *
+     * @param val The value of the counter to be incremented.
+     * @param increment The amount to be added, the value after this incremented must not exceed BufferSize.
+     * @return The new value of the counter.
+     */
+    CounterType incrementCounter(CounterType val, CounterType increment = 1)
     {
-        return (++val) % BufferSize;
+        val += increment;
+
+        MBED_ASSERT(val <= BufferSize);
```

Comment:
this is a private function so we can guarantee we will never exceed buffersize even after increment so we can simplify calculation